packages/cardpay-sdk/sdk/claim-settlement-module.ts

```diff
@@ -19,39 +27,255 @@ export default class ClaimSettlementModule extends SafeModule {
+      return false;
     } else {
-      throw new Error('action not implemented');
+      return false;
+    }
+    return true;
+  }
+
+  async addValidator(txnHash: string): Promise<SuccessfulTransactionReceipt>;
+  async addValidator(
+    moduleAddress: string,
+    avatarAddress: string,
+    validatorAddress: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ): Promise<SuccessfulTransactionReceipt>;
+  async addValidator(
+    moduleAddress: string,
+    avatarAddress?: string,
+    validatorAddress?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ): Promise<SuccessfulTransactionReceipt> {
+    //TODO: Multi-signature. Only supports adding single validator only
+    if (!avatarAddress) {
+      throw new Error('avatarAddress must be specified');
+    }
+    if (!validatorAddress) {
+      throw new Error('validatorAddress must be specified');
+    }
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    let module = new Contract(moduleAddress, this.abi, this.ethersProvider);
+    let data = await module.interface.encodeFunctionData('addValidator', [validatorAddress]);
+    let signer = this.signer ? this.signer : this.ethersProvider.getSigner();
+    let from = contractOptions?.from ?? (await signer.getAddress());
+
+    let safe = new Contract(avatarAddress, GnosisSafeABI, this.ethersProvider);
+    const safeOwners = await safe.getOwners();
+    if (!safeOwners.contain(signer)) {
+      throw new Error(`${signer} is not owner of avatar ${avatarAddress}`);
+    }
+
+    let estimate = await gasEstimate(
+      this.ethersProvider,
+      avatarAddress,
+      module.address,
+      '0',
+      data,
+      Operation.CALL,
+      AddressZero,
+      true
+    );
+
+    if (nonce == null) {
+      nonce = getNextNonceFromEstimate(estimate);
+      if (typeof onNonce === 'function') {
+        onNonce(nonce);
+      }
+    }
+
+    let gnosisTxn = await executeTransaction(
+      this.ethersProvider,
+      avatarAddress,
+      module.address,
+      data,
+      Operation.CALL,
+      estimate,
+      nonce,
+      await signSafeTx(
+        this.ethersProvider,
+        avatarAddress,
+        module.address,
+        data,
+        Operation.CALL,
+        estimate,
+        nonce,
+        from,
+        this.signer
+      )
+    );
+
+    let txnHash = gnosisTxn.ethereumTx.txHash;
+    if (typeof onTxnHash === 'function') {
+      await onTxnHash(txnHash);
+    }
+    return await waitUntilTransactionMined(this.ethersProvider, txnHash);
+  }
+
+  async defaultClaim(moduleAddress: string, payeeAddress: string) {
+    let id = utils.hexlify(utils.randomBytes(32));
+    let startBlockNum = await this.ethersProvider.getBlockNumber();
+    let startBlockTime = (await this.ethersProvider.getBlock(startBlockNum)).timestamp;
+    let validitySeconds = 86400; //1 day
+    let tokenAddress = '0x95093b8836ED53B4594EC748995E45b0Cd2b1389'; // CTST goerli
```

Comment:
I think we can use the getAddress function for this.

Reply:
Resolved [here](https://github.com/cardstack/cardstack/pull/3641/commits/fc47bf612eef2f5f2681a9f1b6ec7244a5c18b93)